test/integration/protocol_integration.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "test/integration/http_integration.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+struct ProtocolTestParams {
```

Comment:
WDYT about `HttpProtocolTest`, http_protocol_integration.h, etc.? We have many different protocols in Envoy.

Reply:
Sure thing!  I'd been waffling about just colocating this with the integration test as it's not a lot of code, but forgot to ask if anyone had opinions.  HttpProtocol SGTM.